extensions/amp-carousel/0.1/build-dom.js

```diff
@@ -0,0 +1,351 @@
+import {isAmp4Email} from '#core/document/format';
+import {isServerRendered} from '#core/dom';
+import {escapeCssSelectorIdent} from '#core/dom/css-selectors';
+import {realChildElements} from '#core/dom/query';
+
+/**
+ * @enum {string}
+ */
+export const ClassNames = {
+  // Carousel Controls
+  BUTTON: 'amp-carousel-button',
+  PREV_BUTTON: 'amp-carousel-button-prev',
+  NEXT_BUTTON: 'amp-carousel-button-next',
+  HAS_CONTROL: 'i-amphtml-carousel-has-controls',
+  CONTROL_HIDE_ATTRIBUTE: 'i-amphtml-carousel-hide-buttons',
+
+  // Generic
+  SLIDE: 'amp-carousel-slide',
+
+  // SlideScroll Carousel
+  SLIDESCROLL_CAROUSEL: 'i-amphtml-slidescroll',
+  SLIDE_WRAPPER: 'i-amphtml-slide-item',
+  SLIDES_CONTAINER: 'i-amphtml-slides-container',
+  SLIDES_CONTAINER_NOSNAP: 'i-amphtml-slidescroll-no-snap',
+
+  // Scrollable Carousel
+  SCROLLABLE_CONTAINER: 'i-amphtml-scrollable-carousel-container',
+  SCROLLABLE_SLIDE: 'amp-scrollable-carousel-slide',
+};
+
+/**
+ * Throws if any provided param is not truthy.
+ */
+function assertDomQueryResults() {
```

Comment:
do you think this would be re-usable in other `build-dom.js` files?

Reply:
It totally is usable in all 3 cases, but as I responded [above](https://github.com/ampproject/amphtml/pull/36602#discussion_r742317260), I'd like to move it to a followup PR which applies this targeted change to a few components.